internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url.go

```diff
@@ -134,7 +137,9 @@ func getResource[T resource.Resource](ctx context.Context, r state.State, namesp
 
 			eventResource, err := event.Resource()
 			if err != nil {
-				watchErr = fmt.Errorf("%s; invalid resource in wrapped event: %w", watchErr.Error(), err)
+				watchErr = multierror.Append(watchErr, fmt.Errorf("invalid resource in wrapped event: %w", err))
```

Comment:
I totally don't understand what's going on in this code (not about your fixes), but more in general why we even have this `watchErr` in the first place